### DISCUSSION TITLE
Need help to know which scalar integrals apply to terms in TID

### DISCUSSION THREAD

**Discussion by Pholipo** [FeynCalc/feyncalc#278] [Q&A] [answered]
Greetings,

I have the following output, named "ex4", this output is the numerator of a loop integral
![image](https://github.com/user-attachments/assets/9cbb4301-33e4-45a1-b583-c5a02c93ccb5)
Where l^{mu} is the loop momentum obtained after Feynman parametrization, and p_{\mu} is the external momentum, this integral is a bubble correction for a massive vector field, with consisting of 2  3-vertex self interactions. After obtaining "ex4", i apply the TID function to it and select the different tensorial contributions to it
![image](https://github.com/user-attachments/assets/68a830d4-d0d4-49f5-9dba-5ec1c8d35a94)
So each of the terms in "ex5" must be multiplied by a scalar loop integral in momentum l. But in "ex4", there are many different powers of l (sometimes l^{2} and a few l^{4}) , and thus they should produce different scalar integrals, how can I tell which scalar integral multiplies which term in the output named "ex5"?

**Comment by vsht** [comment, marked as answer]
Hi,

> So each of the terms in "ex5" must be multiplied by a scalar loop integral in momentum l. But in "ex4", there are many different powers of l (sometimes l^{2} and a few l^{4}) , and thus they should produce different scalar integrals, how can I tell which scalar integral multiplies which term in the output named "ex5"?

sorry, I don't understand the question. The output of TID contains scalar integrals multiplied by the tensor structures w.r.t. which you are collecting. But your `ex5` output looks very weird, because it lacks any propagator denominators. 

BTW, posting pictures instead of expressions is not a good way to ask for meaningful answers, because no one will be able to evaluate your code.

**Reply by Pholipo** [reply]
Thanks, It really lacked the loop denominator!
